### PR TITLE
French revolutionary face

### PR DESCRIPTION
**Pull request by CarpeNoctem**
Initial submission of French Revolutionary (Decimal) time face. ~~I've only run this on the emulator, but have just loaded it onto my watch for testing as well.~~ I'll try to add the power-saving stuff to the output in the coming week or so.

### French Revolutionary Decimal Time

Similar to the Decimal Time face, but with the day divided into ten hours instead of twenty four.
Each hour is divided into one hundred minutes, and those minutes are divided into 100 seconds.
I came across this one the Svalbard watch site here: https://svalbard.watch/pages/about_decimal_time.html
More info here as well: https://en.wikipedia.org/wiki/Decimal_time

### Usage

By default, the face just displays the current decimal time. Pressing the alarm button will toggle through other display options:
1) Just decimal time (with dT indicator at top)
2) Decimal time, with dT indicator and date above.
3) Decimal time, with 24-hr time above (where Day and Date would normally be displayed), BUT minutes first then hours.
   Sadly, the first character of the date area only goes up to 3. (see https://www.sensorwatch.net/docs/wig/display/#the-day-digits) I was going to begrudgindly leave this display option out when I realized that, but thought it would be better to have this backwards representation of the standard time than not at all.

A long-press of the light button will toggle the upper time between 12-hr AM/PM and 24-hr mode. I thought of reading the main setting for this, but thought that a person could normally prefer 12hr time, but next to a 10hr day want to see the normal time in the 24hr format.

A long-press of the alarm button will toggle the seconds off and on.

### CODE REVIEW

**Comment by CarpeNoctem** [comment]
Seems to be running just fine on the watch. The main thing I hadn't been able to test in the emulator was low power mode.

File: movement/watch_faces/clock/french_revolutionary_face.c [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+            else { watch_display_string("--", 8); }
+            break;
+        case EVENT_LIGHT_LONG_PRESS:
+            // In case anyone really wants that upper time in 12-hour format. I thought about using the global setting (settings->bit.clock_mode_24h)
+            // for this preference, but thought someone who prefers 12-hour format normally, might prefer 24hr when compared to a 10hr decimal day,
+            // so this is separate for now.
+            state->use_am_pm = !state->use_am_pm;
+            if (state->use_am_pm) {
+                watch_clear_indicator(WATCH_INDICATOR_24H);
+                if (date_time.unit.hour < 12) { watch_clear_indicator(WATCH_INDICATOR_PM); }
```
Comment (WesleyAC) [inline movement/watch_faces/clock/french_revolutionary_face.c:110]: `date_time` is being used uninitialized here, right? I don't think this is working the way it's intended

File: movement/watch_faces/clock/french_revolutionary_face.c [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include "french_revolutionary_face.h"
+
+void french_revolutionary_face_setup(movement_settings_t *settings, uint8_t watch_face_index, void ** context_ptr) {
+    (void) settings;
```
Comment (WesleyAC) [inline movement/watch_faces/clock/french_revolutionary_face.c:30]: you should also do `(void) watch_face_index;` here to silence the warning about it being unused.

**Review by WesleyAC — CHANGES_REQUESTED** [review]
Very cool! Just a couple changes needed :)

**Comment by CarpeNoctem** [comment]
> Very cool! Just a couple changes needed :)

Thanks so much for this and all of the other reviews, @WesleyAC !

I'll make those changes soon. 👍

File: movement/watch_faces/clock/french_revolutionary_face.c (hunk 2 above)
Comment (CarpeNoctem) [reply movement/watch_faces/clock/french_revolutionary_face.c:30]: Thanks! I'll check whether this has been updated in the new face Python script / template ad well, and try to update that if it's not in yet and there's not already a PR for it.

**Comment by CarpeNoctem** [comment]
Well, that took way longer to get around to than it should have. All fixed up, and upstream merged! Thanks again, @WesleyAC. Great catches.

File: movement/watch_faces/clock/french_revolutionary_face.c (hunk 2 above)
Comment (CarpeNoctem) [reply movement/watch_faces/clock/french_revolutionary_face.c:30]: > Thanks! I'll check whether this has been updated in the new face Python script / template ad well, and try to update that if it's not in yet and there's not already a PR for it.

It had not, so I've added it in the template in PR #360.

**Review by theAlexes — APPROVED** [review]

**Review by matheusmoreira — APPROVED** [review]

**Comment by matheusmoreira** [comment]
This PR has been merged into `main` but did not automatically close because it was rebased.

**Pull request closed without merging**